src/classes/WalletManager/index.ts

```diff
@@ -37,19 +37,22 @@ export class WalletManager {
     private _wrappedProvider: BrowserProvider | null = null
     private _nativeProvider: MetaMaskEthereumProvider | null = null
     private _walletInfo: WalletInfo | null = null
-    private _currentWalletApplication: WalletApplication | null = null
+    private _currentWalletConnectorId: string | null = null
     private _initializedEvents = false
+    private _connectors: Record<string, WalletConnector> = {}
 
     /**
      * Constructs a WalletProvider instance.
      *
      * @param defaultChainId - The default blockchain network to connect to.
      * @param availableChainIds - An array of available blockchain networks for connection.
+     * @param _connectors - An object of wallet connectors mapped by their IDs.
      * @param _onWalletUpdate - Callback to be executed on wallet updates, such as a change in the address, chain, or a disconnect.
      */
     constructor(
         public readonly defaultChainId: ChainId,
         public readonly availableChainIds: ChainId[],
+        connectors: WalletConnector[],
         private readonly _onWalletUpdate?: OnWalletUpdate,
```

Comment:
connectors should be optional and default to all available inside the library if not passed.

also, since we will have 2 optional params maybe we should pass an object so we wont have issues calling this constructor passing only the second param